### PR TITLE
New user guide / tutorials

### LINKED ISSUE #1108: Sklearn-style user guide
We have previously discussed an sklearn-style user guide, that goes into some more mathematical depth, and provides a translation of math <-> code. Let's use this issue/corresponding PR to discuss a standardized template.

### PR DESCRIPTION
**Pull request by adamgayoso**
Closes #1108

This is the initial skeleton for the new docs. We can merge this and then continue to make PRs for the rest of the models.

### CODE REVIEW

**Comment by watiss** [comment]
@adamgayoso Can you send a link to a draft version of the site generated from these changes?

**Comment by adamgayoso** [comment]
@watiss https://scvi--1127.org.readthedocs.build/en/1127/user_guide/index.html

You can also find it in the checks of the PR, via the readthedocs check, after it completes.

File: docs/user_guide/models/cellassign.rst [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+========================
+CellAssign uses expectation maximization (EM), which optimizes its parameters and provides a cell-type prediction for each cell.
+
+Initialization
+--------------
+
+CellAssign initializes parameters as follows:  :math:`\delta_{gc}` is initialized with a :math:`\textrm{LogNormal}(0, 1)`
+draw, :math:`\bar{\delta}` is initialized to 0, :math:`\sigma^2` is initialized to 1, :math:`\pi_i` is
+initialized to :math:`1/C`, :math:`\beta_{gi}` is initialized with a :math:`\mathcal{N}(0, 1)` draw,
+and the inverse dispersion parameter :math:`\log a_i` is initialized to zero. Additionally, :math:`b` is fixed a priori to be
```
Comment (jjhong922) [inline docs/user_guide/models/cellassign.rst:86]:
```suggestion
and the inverse dispersion parameter :math:`\log a_i` is initialized to 0. Additionally, :math:`b` is fixed a priori to be
```

File: docs/user_guide/models/cellassign.rst [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+in the original implementation [#ref2]_, using the Adam optimizer [#ref3]_, except that now an optimization step corresponds to data from one minibatch. Following the original implementation, we
+also clamped :math:`\delta > 2`. We also added early stopping with respect to the log likelihood of a held-out validation set.
+
+Tasks
+=====
+
+Cell type prediction
+---------------------
+
+The primary task of CellAssign is to predict cell types for each cell. This is accomplised by::
```
Comment (jjhong922) [inline docs/user_guide/models/cellassign.rst:140]:
```suggestion
The primary task of CellAssign is to predict cell types for each cell. This is accomplished by::
```

File: docs/user_guide/models/cellassign.rst [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+    >>> model = CellAssign(adata, marker_gene_matrix, size_factor_key='size_factor')
+    >>> model.train()
+    >>> predictions = model.predict(adata)
+
+where `predictions` stores :math:`\gamma_{nc}` for each cell :math:`n` and cell type :math:`c`.
+
+
+Implementation details
```
Comment (jjhong922) [inline docs/user_guide/models/cellassign.rst:149]: Given the number of variables here, I think a table like in the scVI guide would be helpful.

File: docs/user_guide/models/totalvi.rst [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+
+
+Preliminaries
+==============
+totalVI takes as input a scRNA-seq gene expression matrix of UMI counts :math:`X` with :math:`N` cells and :math:`G` genes
+along with a paired matrix of protein abundance (UMI counts) :math:`Y`, also of :math:`N` cells, but with :math:`T` proteins.
+Thus, for each cell, we observe both RNA and protein information.
+Additionally, a design matrix :math:`S` containing :math:`p` observed covariates, such as day, donor, etc, is an optional input.
+While :math:`S` can include both categorical covariates and continuous covariates, in the following, we assume it contains only one
+categorical covariate with :math:`B` categories, which represents the common case of having multiple batches of data.
```
Comment (jjhong922) [inline docs/user_guide/models/totalvi.rst:34]: The number of categories was K in the scVI guide, should we be consistent?

File: docs/user_guide/models/totalvi.rst [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+
+
+Generative process
+========================
+
+.. figure:: figures/totalvi_graphical_model.svg
```
Comment (jjhong922) [inline docs/user_guide/models/totalvi.rst:40]: would suggest moving this further below once you have defined the variables

Comment (jjhong922) [reply docs/user_guide/models/totalvi.rst:40]: also noticed that this image doesnt include the global parameters

**Review by jjhong922 — APPROVED** [review]
Left a bunch of nits, but overall the structure of stuff looks nice. Also, if there is a way to have a hierarchical structure on the outline on the left sidebar, it would be nice to match it with the User Guide landing page.

File: docs/user_guide/models/totalvi.rst (hunk 5 above)
Comment (adamgayoso) [reply docs/user_guide/models/totalvi.rst]: going to leave this as is, we typically haven't included global params in these graphical models

File: docs/user_guide/models/totalvi.rst (hunk 4 above)
Comment (adamgayoso) [reply docs/user_guide/models/totalvi.rst:34]: probably yes

File: docs/user_guide/background/variational_inference.rst [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+
+
+In addition to the model parameters :math:`\theta`, let :math:`\phi` be the parameters of the approximate posterior (e.g., mean and covariance matrix if :math:`Q` is
+the family of multivariate Gaussians). Using the derivation above, we can express the log probability of the data as
+
+.. math::
+   :nowrap:
+
+    \begin{align}
+        p_\theta(x) &= \mathrm{KL}(\,q_\phi(z)\, \| \,p_\theta(z \mid x)\,) + \mathbb E_{q} \left[\log p_\theta(x \mid z)\right] -  \mathrm{KL}(\,q_\phi(z)\, \| \,p_\theta(z)\,)\\
```
Comment (watiss) [inline docs/user_guide/background/variational_inference.rst:84]: should it be log p theta(x) ?

File: docs/user_guide/background/variational_inference.rst [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+Amortizing inference
+=====================
+
+Let's think about our high-dimensional single-cell data. If we have a model where we want to learn a posterior for a low-dimensional latent variable :math:`z`, then there will be an approximate posterior
+mean and variance parameter associated with each single cell. This scales linearly with the number of cells in our dataset.
+
+To amortize the cost of inference, we can tie together all cells using a neural network. Now suppose that we feed each cell's gene expression :math:`x` through a neural network that outputs the mean and variance
+of :math:`q(z)` for that cell. Inference will tend to be more scalable as the variational parameters are global (shared by all cells). On the other hand, we may pay a cost called the *amortization gap* [#ref2]_ for this sharing.
+
+Overall, this technique is typicall called *amortized variational inference* and in some cases *auto-encoding variational bayes* [#ref3]_.
```
Comment (watiss) [inline docs/user_guide/background/variational_inference.rst:121]: typically*

**Review by watiss — APPROVED** [review]

File: docs/user_guide/models/cellassign.rst [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+like batch or donor. The scvi-tools implementation of CellAssign uses stochastic inference,
+such that CellAssign will scale to very large datasets.
+
+The advantages of CellAssign are:
+
+    + Lightweight model that can be fit quickly.
+
+    + Ability to control for nuisance factors.
+
+The disadvantages of CellAssign include:
```
[the remark this answers is not in the document]
Comment (adamgayoso) [reply docs/user_guide/models/cellassign.rst:17]: good idea

File: docs/user_guide/models/scvi.rst [hunk 9]
```diff
@@ -0,0 +1,198 @@
...
+Data simulation
+---------------
+
+Data can be generated from the model using the posterior predictive distribution in :func:`~scvi.model.SCVI.posterior_predictive_sample`.
+This is equivalent to feeding a cell through the model, sampling from the posterior
+distributions of the latent variables, retrieving the likelihood parameters (of :math:`p(x \mid z, s)`), and finally, sampling from this distribution.
+
+
+
+.. topic:: References:
```
[the remark this answers is not in the document]
Comment (adamgayoso) [reply docs/user_guide/models/scvi.rst:193]: I would say that scvi-tools encompasses all the methods. Best to make the original method called scVI seem independent so others don't think including their methods in scvi-tools means that it's scVI.

File: docs/user_guide/index.rst [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+   :widths: 15 15 15 100
+   :header-rows: 1
+
+   * - Model
+     - Reference
+     - Class
+     - Tasks
+   * - :doc:`/user_guide/models/scvi`
+     - [Lopez18]_
+     - :class:`~scvi.model.SCVI`
```
[the remark this answers is not in the document]
Comment (adamgayoso) [reply docs/user_guide/index.rst]: done

File: docs/user_guide/index.rst [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+     - Dimensionality reduction, removal of unwanted variation, integration across replicates, donors, and technologies, differential expression, imputation, normalization of other cell- and sample-level confounding factors
```
[the remark this answers is not in the document]
Comment (adamgayoso) [reply docs/user_guide/index.rst:17]: need to figure this out, will post in the issue.

**Pull request merged**